lib/zip/entry.rb

```diff
@@ -406,24 +406,28 @@ def get_extra_attributes_from_path(path) # :nodoc:
       @unix_uid   = stat.uid
       @unix_gid   = stat.gid
       @unix_perms = stat.mode & 0o7777
+      @time = ::Zip::DOSTime.from_time(stat.mtime)
```

Comment:
- This is currently guarded by `return if Zip::RUNNING_ON_WINDOWS`. I am pretty sure that the modification time would also be available on Windows, and from what I can see in the documentation, `File::stat` should still work. I have been meaning to find a Windows machine to test this on... I guess even if the `restore_times` only works on Linux, it would still be an improvement.

- What do you think to using `time=` rather than assigning the `@time` instance variable directly? It looks like that would also set some related attributes in `@extra`, which seems desirable.

Reply:
Good point, I will try using `time=`. TBH working through how all this works was horribly convoluted so I must have just missed this subtlety 😬

Reply:
Urgh. So I don't think the `UniversalTime` extra field can work as currently implemented. In `Entry#time=` when it creates a `UniversalTime` entry in `@extra` the `.flag` field is set to `nil`. But the `UniversalTime` field can only be packed if `@flag` is non-`nil`.

Going through the tests for the extra field code it seems that `@flag` is always set to 3 for `UniversalTime`, and indeed forcing this does work. But I need to look and see if this reasonable.

It's late now, so I'll pick this up another time.

Reply:
Right. I think given the issues with using `time=` at the moment, I think it's best to keep using `@time` for now.

I've found documentation for the `UniversalTime` extra field now, so I think I can fix it in a different PR once I've properly got my head round it.

I also think, now I've seen what adding the extra fields does to the archive, that having a way of *not* using extra fields by default (as suggested in #398) would be a good idea. So I'll have a think about that too.

I'll finish up making the other changes to this PR we've already discussed this evening and update it.